hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -398,6 +417,19 @@ private Target createTarget(final String controllerId, final URI address) {
         return result;
     }
 
+    private Target createTarget(final String controllerId, final URI address, final String name) {
```

Comment:
You could avoid code duplication in the createTarget methods by calling  `return createTarget(controllerId, address, controllerId);`  inside crateTarget(String, URI)

Reply:
Acutally one method createTarget(String, URI, String) could be sufficient here, considering the comment for `findOrRegisterTargetIfItDoesNotExist`

Reply:
Changed it to two methods, one called createTargetWithName(), and the other called createTarget(). Still I would not combine them because of separation of concerns concept and readability.